lecture_5/homework/app/services/reports/siege_report.rb

```diff
@@ -3,10 +3,29 @@
 module Reports
   class SiegeReport
     def initialize(building:)
+      @building = building
     end
 
     def call
-      raise NotImprementedYet
+      calc_siege_ability
+    end
+
+    private
+
+    attr_reader :building
+
+    def calc_siege_ability
+      return 0 if @building.warriors.count.zero?
```

Comment:
To sprawdzenie spokojnie by się mogło znaleźć gdzieś wyżej, żeby nie obciążać logiki serwisu :) Serwis najlepiej jeśli ma jedną odpowiedzialność, a Twój serwis poza liczeniem ile dni przetrwa dany budynek, stwierdza upadłość budynku jeśli nie ma w nim wojowników :)